File: projekt_java/src/po/java/zwierzeta/Czlowiek.java
package po.java.zwierzeta;

import po.java.Organizm;
import po.java.Swiat;
import po.java.Zwierze;

import java.awt.*;
import java.util.ArrayList;
import java.util.Random;

public class Czlowiek extends Zwierze {
    private int kierunek,bron;
    public void setKierunek(int kierunek) {
        this.kierunek = kierunek;
    }
    public int getBron() {
        return bron;
    }
    public void setBron(int bron) {
        this.bron = bron;
    }
    public Czlowiek(int x, int y)
    {
        this.x = x;
        this.y = y;
        this.xprev = x;
        this.yprev = y;
        this.symbol = 'C';
        this.sila = 5;
        this.inicjatywa = 4;
        this.zycie = 0;
        this.wykonal_ruch=true;
        this.numer=5;
        this.bron=-5;
        this.kolor= Color.PINK;
    }
    public Czlowiek(int []dane)
    {
        this.x = dane[0];
        this.y = dane[1];
        this.xprev = dane[2];
        this.yprev = dane[3];
        this.inicjatywa=dane[4];
        this.symbol = 'C';
        this.sila = dane[5];
        this.wykonal_ruch = false;
        this.zycie = dane[6];
        this.numer=dane[7];
        this.bron=dane[8];
        this.kolor= Color.PINK;
    }

    @Override
    public void wykonajRuch(Swiat swiat, ArrayList organizmy, int j) {
        if (kierunek == 0)//^
        {
            if(x>0)
                x--;
        }
        if (kierunek == 1)//>
        {
            if(y<swiat.getRozmiary()-1)
                y++;
        }
        if (kierunek == 2)//v
        {
            if(x<swiat.getRozmiarx()-1)
                x++;
        }
        if (kierunek == 3)//<
        {
            if(y>0)
                y--;
        }
    }
    @Override
    public void kolizja(Organizm organizm, ArrayList organizmy, int j, Swiat swiat) {
        if (bron > 0&&organizm instanceof Zwierze)
        {
            int kierunek,xact,yact;
            Random liczba=new Random();
                do
                {
                    xact = organizm.getX();
                    yact = organizm.getY();
                    kierunek = liczba.nextInt(4);

                    if (kierunek == 0)//^
                    {
                        xact --;
                    }
                    if (kierunek == 1)//>
                    {
                        yact ++;
                    }
                    if (kierunek == 2)//v
                    {
                        xact ++;
                    }
                    if (kierunek == 3)//<
                    {
                        yact --;
                    }
                    if (swiat.czyWPlanszy(swiat, xact, yact)&&swiat.getPlansza(xact,yact)==null)  break;
                } while (true);

            swiat.setRaport("C odpycha "+organizm.getSymbol());
            organizm.setX(xact);
            organizm.setY(yact);
            swiat.setPlanszaKolizja(organizm.getX(), organizm.getY(), organizm);
            return;
        }
        if (organizm.getSila()>sila)
        {
            organizmy.remove(this);
            swiat.setRaport("C ginie od "+organizm.getSymbol());
            swiat.setPlanszaKolizja(x, y, organizm);
            j--;
        }
        else
        {
            organizmy.remove(organizm);
            swiat.setRaport(organizm.getSymbol()+" ginie od C");
            j--;
        }
    }
    @Override
    public String toString()
    {
        return symbol+" "+x+" "+y+" "+xprev+" "+yprev+" "+inicjatywa+" "+sila+" "+zycie+" "+numer+" "+bron;
    }
}

File: Projekt_Python/zwierzeta/Czlowiek.py
import random
from Zwierze import Zwierze


class Czlowiek(Zwierze):

    def __init__(self, x, y, dane):
        if dane == None:
            self._x = x
            self._y = y
            self._xprev = x
            self._yprev = y
            self._sila = 5
            self._inicjatywa = 4
            self._zycie = 0
            self._wykonal_ruch = True
            self._numer = 6
            self.__bron = -5
        else:
            self._x = dane[0]
            self._y = dane[1]
            self._xprev = dane[2]
            self._yprev = dane[3]
            self._inicjatywa = dane[4]
            self._sila = dane[5]
            self._wykonal_ruch = False
            self._zycie = dane[6]
            self._numer = dane[7]
            self.__bron = dane[8]
        self.__kierunek = -1
        self._symbol = 'C'
        self._kolor = (250, 128, 114)

    def setKierunek(self, kierunek):
        self.__kierunek = kierunek

    def getKierunek(self):
        return self.__kierunek

    def getBron(self):
        return self.__bron

    def setBron(self, bron):
        self.__bron = bron

    def wykonajRuch(self, swiat, organizmy, j):
        if self.__kierunek == 0:  # //^
            if self._x > 0:
                self._x -= 1
        if self.__kierunek == 1:  # //>
            if self._y < swiat.getRozmiary() - 1:
                self._y += 1
        if self.__kierunek == 2:  # //v
            if self._x < swiat.getRozmiarx() - 1:
                self._x += 1
        if self.__kierunek == 3:  # //<
            if self._y > 0:
                self._y -= 1

    def kolizja(self, organizm, organizmy, j, swiat):
        if self.__bron > 0 and isinstance(organizm, Zwierze):
            while True:
                xact = organizm.getX()
                yact = organizm.getY()
                kierunek = random.randint(0, 3)

                if kierunek == 0:  # //^
                    xact -= 1
                if kierunek == 1:  # //>
                    yact += 1
                if kierunek == 2:  # //v
                    xact += 1
                if kierunek == 3:  # //<
                    yact -= 1
                if swiat.czyWPlanszy(xact, yact) and swiat.getPlansza(xact, yact) == None:  break

            swiat.setRaport("C odpycha " + organizm.getSymbol())
            organizm.setX(xact)
            organizm.setY(yact)
            swiat.setPlanszaKolizja(organizm.getX(), organizm.getY(), organizm)
            return

        if organizm.getSila() > self._sila:
            organizmy.remove(self)
            swiat.setRaport("C ginie od " + organizm.getSymbol())
            swiat.setPlanszaKolizja(self._x, self._y, organizm)
            j -= 1
        else:
            organizmy.remove(organizm)
            swiat.setRaport(organizm.getSymbol() + " ginie od C")
            j -= 1

    def __str__(self):
        return "%r, %r, %r,%r, %r, %r, %r, %r, %r, %r" % (
            self._symbol, self._x, self._y, self._xprev, self._yprev, self._inicjatywa, self._sila, self._zycie,
            self._numer, self.__bron)
